### ISSUE TITLE
Support backend.run() in IBM Cloud channel 

### ISSUE THREAD

**Issue by kt474** [Qiskit/qiskit-ibm-runtime#1117]
We are moving the Researchers program from IQP to IBM Cloud. Since researchers are likely to be interested in functionalities provided by backend.run (e.g. dynamic circuits), we want to support `backend.run()` on the IBM Cloud channel as well.

Since `backend.run` [inputs are changing](https://github.com/Qiskit/qiskit-ibm-provider/issues/640), `backend.run` in `qiskit-ibm-runtime` should just accept new inputs only. People who want to use the old can continue to use `qiskit-ibm-provider`. 

Note that `backend.run()` may become a third primitive, and the interface may change

**Comment by jyu00** [comment]
Yes, [the epic](https://github.ibm.com/IBM-Q-Software/requirements-and-priorities/issues/296) has more details. 

tldr, it'll just be like `backend.run` in `qiskit-ibm-provider`. We may wrap it as a primitive eventually, but it'd still need to go through a deprecation period if/when that happens.
